### PR TITLE
Updating to new api

### PR DESCRIPTION
**Pull request by ghop02**
Updates all references to point to API v2 files.  

The biggest change is to a lot of the types in the new API.  A lot more values can be undefined. This was a bug in the original json spec, where it said that values would be required in the json schema spec, but they weren't actually required.  

Also, `stateName` was replaced with `state`, so the lookup goes the other direction now.  

I need to add a PR in covid data model to get the latest json-schema version on master (right now it points to a specific commit)

### CODE REVIEW

File: src/common/models/Projection.ts
Comment (mikelehen) [inline src/common/models/Projection.ts:146]: FWIW- I try to use `stateCode` or `stateId` since `state` is kinda' ambiguous.

File: src/common/models/Projection.ts
Comment (mikelehen) [inline src/common/models/Projection.ts:205]: As per before, this and all the similar constructs below will break when cases is `0`.

Not sure what prompted all these changes but if we need to tweak the types to use `undefined` (or even `undefined|null`) for missing values, that would probably be fine...

File: src/common/models/Projection.ts
Comment (mikelehen) [inline src/common/models/Projection.ts:294]: Should we drop this override since it's not used, and it won't work right unless we override it in the backend and the frontend anyway, which is 🤮 ?

File: src/common/models/Projection.ts
Comment (mikelehen) [inline src/common/models/Projection.ts:434]: Can you remove this TODO?  It's outdated (at some point the API started including 'County' in the names...)

File: src/common/models/Projection.ts
Comment (mikelehen) [inline src/common/models/Projection.ts:437]: Perhaps add a getStateName() helper to locations.ts that either does the ts-ignore or `(STATES as any)[stateCode]`.

Alternatively, we have [getLocationNameForFips()](https://github.com/covid-projections/covid-projections/blob/062a2ac10fae43292e5c42eb34f48fbc69ab9f21/src/common/locations.ts#L119) we could use, but it only includes the state abbreviation.  We could change it to accept options or something like `getLocationNameForFips(s.fips, { useStateName: true })`

File: src/common/utils/model.ts
Comment (mikelehen) [inline src/common/utils/model.ts:52]: Hrm...  I think the filter you removed is what allowed us to add territories in the backend without breaking the frontend.

File: src/common/utils/model.ts
Comment (mikelehen) [inline src/common/utils/model.ts:73]: ditto

**Review by mikelehen — CHANGES_REQUESTED** [review]
Mostly looks good but a few nits and I think we ended up with some problematic `0` handling...

File: scripts/update_api_types.ts
Comment (ghop02) [reply scripts/update_api_types.ts:41]: fixed!

File: src/common/models/Projection.ts
Comment (ghop02) [reply src/common/models/Projection.ts:205]: ah this makes sense.  Yeah - a few days ago I tried to tweak the types to use `undefined` and it turned into a lot bigger change set than originally intended.  My immediate reaction is just to use [isNil](https://lodash.com/docs/#isNil) to check for nil-ish values for all these cases. Does that seem alright? or happy to go down a different path also

File: src/common/utils/model.ts
Comment (ghop02) [reply src/common/utils/model.ts:52]: yeah that makes sense. I think I need to check if the state code is in the states object instead (because now you'll never have a null state code, but could run into problems elsewhere)

**Review by mikelehen — APPROVED** [review]
Mostly looks good but a few cases still need improved null/undefined handling I think.  I also didn't get a chance to test this, so please be careful.  Or if this isn't urgent, I can give it more attention when I'm back.

File: src/common/models/Projection.ts
Comment (ghop02) [reply src/common/models/Projection.ts]: Yeah that makes sense. 

Ended up fixing the typing (https://github.com/covid-projections/covid-data-model/pull/710) here and simplifying this PR a lot

File: src/common/models/Projection.ts
Comment (ghop02) [reply src/common/models/Projection.ts]: yup!

File: src/common/models/Projection.ts
Comment (ghop02) [reply src/common/models/Projection.ts]: this is fixed with better api typing!

File: src/common/models/Projection.ts
Comment (ghop02) [reply src/common/models/Projection.ts]: Dropped!

File: src/common/models/Projection.ts
Comment (ghop02) [reply src/common/models/Projection.ts]: removed

File: src/common/utils/snapshots.ts
Comment (mikelehen) [inline src/common/utils/snapshots.ts:17]: I noticed that https://covid-projections-git-updating-to-new-api.covidactnow.vercel.app/internal/compare isn't working and I think it's because we end up getting the current snapshot URL from _master_ (see fetchMasterSnapshotNumber() above) which doesn't include /v2/...  I think the email alerts may do the same thing.  Should we make this regex accept either the old or new snapshot URLs?  Else, depending on how we deploy things we may have a temporary breakage?

**Review by mikelehen — APPROVED** [review]
Nice! This looks good with just a couple small tweaks.

File: src/assets/data/data_url.json
Comment (ghop02) [reply src/assets/data/data_url.json:2]: Yup, was going to wait until today's build is merged before checking

**Pull request merged**
